### PR TITLE
create unit tests for DefaultBcmcDelegate.kt

### PR DESCRIPTION
**Pull request by etman55**

### CODE REVIEW

File: bcmc/build.gradle [hunk 1]
```diff
@@ -30,18 +30,25 @@ android {
...
+    testOptions {
+        unitTests.returnDefaultValues = true
+    }
 }
 
 dependencies {
     // Checkout
     api project(":card")
-
+    api project(':cse')
```
Comment (jreij) [inline bcmc/build.gradle:42]: Doesn't `card` include `cse` anyway?

File: bcmc/src/test/java/com/adyen/checkout/bcmc/DefaultBcmcDelegateTest.kt [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+                }
+            }
+        }
+
+        @Test
+        fun `card number in output data is invalid, then component state should be invalid`() = runTest {
+            delegate.initialize(CoroutineScope(UnconfinedTestDispatcher()))
+            delegate.componentStateFlow.test {
+                delegate.createComponentState(
+                    BcmcOutputData(
```
Comment (jreij) [inline bcmc/src/test/java/com/adyen/checkout/bcmc/DefaultBcmcDelegateTest.kt:200]: Can you add names to call arguments?

File: bcmc/src/test/java/com/adyen/checkout/bcmc/DefaultBcmcDelegateTest.kt [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+                with(requireNotNull(expectMostRecentItem())) {
+                    assertTrue(isValid)
+                    assertTrue(isInputValid)
+                }
+            }
+        }
+    }
+
+    private fun createOutputData(): BcmcOutputData {
+        return BcmcOutputData(
```
Comment (jreij) [inline bcmc/src/test/java/com/adyen/checkout/bcmc/DefaultBcmcDelegateTest.kt:250]: Can you add names to call arguments?

File: bcmc/build.gradle (hunk 1 above)
Comment (etman55) [reply bcmc/build.gradle:42]: Yes, you're right

**Pull request closed without merging**